### ISSUE TITLE
ClassConstructorAttribute not working at assembly level

### ISSUE THREAD

**Issue by Barsonax** [thomhurst/TUnit#1371]
First of all nice framework, love the source gen approach and how fast it is and the very flexible way everything can be setup. Makes TUnit a great fit for any kind of test!

Since it is allowed to put `ClassConstructorAttribute` at the assembly level I would expect it to work like an default for the entire assembly meaning if there is no `ClassConstructorAttribute` defined at the class level the assembly level one should be used. In practice it doesn't seem to do anything.

Is my assumption correct and thus this not working correctly or was this by design and am I missing something?

**Comment by thomhurst** [comment]
Thanks!

Yeah it should work at the assembly level, so for example you could define a generic dependency injection constructor for example.

I'll take a look!

**Comment by thomhurst** [comment]
Hmm... I can't seem to recreate. Mind producing a minimal repro for me?

**Comment by Rick-van-Dam** [comment]
This is the exact error iam seeing:
![image](https://github.com/user-attachments/assets/83e68c5d-9609-43e7-b72a-e2682ad2f425)

And this the minimal test code:
```cs
using System.Diagnostics.CodeAnalysis;
using ClassLibrary1;
using Microsoft.Extensions.DependencyInjection;
using TUnit.Core.Interfaces;

[assembly: ClassConstructor<DependencyInjectionClassConstructor>]

namespace ClassLibrary1;

public class DependencyInjectionClassConstructor : IClassConstructor, ITestEndEventReceiver
{
    private static readonly IServiceProvider ServiceProvider = CreateServiceProvider();

    private AsyncServiceScope _scope;

    public T Create<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] T>(ClassConstructorMetadata classConstructorMetadata)
        where T : class
    {
        _scope = ServiceProvider.CreateAsyncScope();
        return ActivatorUtilities.GetServiceOrCreateInstance<T>(_scope.ServiceProvider);
    }

    public ValueTask OnTestEnd(TestContext testContext) => _scope.DisposeAsync();

    private static ServiceProvider CreateServiceProvider() =>
        new ServiceCollection().AddSingleton<Dependency>().BuildServiceProvider();
}

public class Class1(Dependency dependency)
{
    [Test]
    public void Foo()
    {

    }
}

public class Dependency
{

}
```

And to be complete this is the csproj:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net9.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="9.0.0"/>
    <PackageReference Include="TUnit" Version="0.4.74"/>
  </ItemGroup>

</Project>
```

Its also not just rider but also on the commandline:
```
PS H:\git\MyNetdaemonSmarthome\ClassLibrary1> dotnet test
Restore complete (0.2s)
  ClassLibrary1 failed with 1 error(s) and 2 warning(s) (0.4s)
    H:\git\MyNetdaemonSmarthome\ClassLibrary1\TUnitStartup.cs(29,32): warning CS9113: Parameter 'dependency' is unread.
    H:\git\MyNetdaemonSmarthome\ClassLibrary1\TUnitStartup.cs(29,14): error TUnit0038: No data source provided
    H:\git\MyNetdaemonSmarthome\ClassLibrary1\TUnitStartup.cs(32,17): warning CA1822: Member 'Foo' does not access instance data and can be marked as static (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1822)

Build failed with 1 error(s) and 2 warning(s) in 1.0s

```

**Comment by thomhurst** [comment]
Ah it's an analyzer error, sorry I thought it was a source generator issue at first.

I see the issue 😄 Will get a fix out

**Comment by thomhurst** [comment]
Fix should be in v0.4.86

**Comment by Rick-van-Dam** [comment]
Yup its working now
